packages/splits-sdk/src/client/splitV2.ts

```diff
@@ -621,6 +623,52 @@ export class SplitV2Client extends SplitV2Transactions {
   async owner(split: Address): Promise<Address> {
     return this._getSplitV2Contract(split).read.owner()
   }
+
+  async getSplitMetadata(splitAddress: Address): Promise<{ split: SplitV2 }> {
+    this._requirePublicClient()
+
+    const [updateLogs, createLogs, owner] = await Promise.all([
+      this._publicClient?.getLogs({
+        address: splitAddress,
+        event: splitUpdatedEvent,
+        strict: true,
+      }),
+      this._publicClient?.getLogs({
+        event: splitCreatedEvent,
+        args: {
+          split: splitAddress,
+        },
+        strict: true,
+      }),
+      this.owner(splitAddress),
+    ])
+
+    if (!createLogs) throw new Error('Split not found')
+
+    let split: SplitV2 = {
+      address: splitAddress,
+      recipients: createLogs[0].args.splitParams.recipients as Address[],
+      allocations: createLogs[0].args.splitParams.allocations as bigint[],
+      totalAllocation: createLogs[0].args.splitParams.totalAllocation,
+      distributionIncentive:
+        createLogs[0].args.splitParams.distributionIncentive,
+      type: getSplitType(this._chainId, createLogs[0].address),
+      owner,
+      creator: createLogs[0].args.creator,
+    }
+
+    if (!updateLogs || updateLogs.length == 0) return { split }
+
+    updateLogs.sort((a, b) => (a.blockNumber > b.blockNumber ? -1 : 1))
+
+    split.recipients = updateLogs[0].args._split.recipients as Address[]
+    split.allocations = updateLogs[0].args._split.allocations as bigint[]
+    split.totalAllocation = updateLogs[0].args._split.totalAllocation
+    split.distributionIncentive =
+      updateLogs[0].args._split.distributionIncentive
```

Comment:
I think this is right but just double checking. The UpdateSplit event will emit the full list of addresses/allocations, right? That's why just using the most recent one works, right?

Reply:
yes